crates/api_crud/src/post/update.rs

```diff
@@ -47,26 +46,41 @@ pub async fn update_post(
 ) -> LemmyResult<Json<PostResponse>> {
   let local_site = LocalSite::read(&mut context.pool()).await?;
 
-  // TODO No good way to handle a clear.
-  // Issue link: https://github.com/LemmyNet/lemmy/issues/2287
-  let url = data.url.as_ref().map(clean_url_params);
-  let custom_thumbnail = data.custom_thumbnail.as_ref().map(clean_url_params);
+  let url = diesel_url_update(&data.url)?;
+
+  let custom_thumbnail = diesel_url_update(&data.custom_thumbnail)?;
 
   let url_blocklist = get_url_blocklist(&context).await?;
 
   let slur_regex = local_site_to_slur_regex(&local_site);
-  check_slurs_opt(&data.name, &slur_regex)?;
-  let body = process_markdown_opt(&data.body, &slur_regex, &url_blocklist, &context).await?;
+
+  let body = diesel_string_update(
+    &process_markdown_opt(&data.body, &slur_regex, &url_blocklist, &context).await?,
+  );
+
+  let alt_text = diesel_string_update(&data.alt_text);
 
   if let Some(name) = &data.name {
     is_valid_post_title(name)?;
+    check_slurs(name, &slur_regex)?;
+  }
+
+  if let Some(Some(body)) = &body {
```

Comment:
Can call `flatten` on `body`.

Reply:
I tried, but in all these cases it forces me to do a `.clone()`, in order to not move the value of body, and `.flatten()` doesn't work on refs of options for some reason.